### PR TITLE
(#104) Update code for Discord.js v14

### LINKED ISSUE #104: Upgrade to Discord.js v14
See #97

### PR DESCRIPTION
**Pull request by mkevenaar**
Updates the code to support Discord.js v14

## Related Issue

Fixes #104 
Closes #97 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -14,6 +15,6 @@ export default {
   ],
   globalSetup: './test/setup-global.js',
   setupFilesAfterEnv: [],
-  collectCoverageFrom: ['**/*.js'],
+  collectCoverageFrom: ['**/*.js', '**/*.cjs', '**/*.mjs'],
```
Comment (davidzwa) [inline jest.config.js:18]: Do you have any relevant `.cjs` files in your repo? Just curious 👍🏼

Comment (mkevenaar) [reply jest.config.js:18]: Currently not, but could be forgotten when adding a cjs file in the future. This makes sure that coverage is there

File: src/commands/fun/whois.js [hunk 2]
```diff
@@ -59,37 +61,37 @@ export async function execute(interaction, client) {
...
-    let createDate = Math.floor(user.createdAt / 1000);
+    let createDate = Math.floor(member.user.createdAt / 1000);
 
     // User Flags
     const userFlags = (await member.user.fetchFlags()).toArray();
 
     // Key Permissions
     const keyPermissions = member.permissions.toArray().filter((p) => elevatedPerms.includes(p));
-    if (keyPermissions.includes('ADMINISTRATOR'))
+    if (keyPermissions.includes('Administrator'))
```
Comment (davidzwa) [inline src/commands/fun/whois.js:71]: Hmm this is something I would be careful with concerning typos. I think you are all good on the typos part, just saying.
Maybe use a constant for this to make sure you get warnings in your favorite IDE:

``` 
const Permissions = {
   ADMINISTRATOR: "Administrator"
   etc: "etc"
};

// usage: Permissions.ADMINISTRATOR
```

Thoughts?

Comment (mkevenaar) [reply src/commands/fun/whois.js:71]: Improvement --> own issue?
Perhaps there are named constants already in Discort.JS. Haven't checked

File: test/application/bot.test.js.disabled [hunk 3]
```diff
@@ -1,3 +1,7 @@
+/**
+ * Disabled due to issues with Discord.JS v14.
+ * TODO: rework this so it will work with Discord.JS v14!
```
Comment (davidzwa) [inline test/application/bot.test.js.disabled:3]: Not a comment for change, just an FYI. My analysis is that the test is doing 'async work' after the Client is constructed. Bad practice? Yes. Reality? Also yes.

This is a problem we could solve by mocking. It's not easy and takes quite some Jest digging. Your comment is 100% justified here.

**Review by davidzwa — APPROVED** [review]
The work in this PR is truly showing you have grown as a JS dev. Well done.
#109 is a unrelated spinoff issue. 

I must add: I have not tested it functionally and I have no experience with v14 whatsoever, but the changes make perfect sense with what they say concerning breaking changes. I trust you will test it through time/bug reports.

File: test/application/bot.test.js.disabled (hunk 3 above)
Comment (mkevenaar) [reply test/application/bot.test.js.disabled:3]: I think we should create an issue for this, so we won’t forget!

Comment (mkevenaar) [reply test/application/bot.test.js.disabled:3]: #110

**Comment by davidzwa** [comment]
Based on DM, I suggest this is merged. The integration test issues will be tracked in https://github.com/mkevenaar/CharlieSpring/issues/110#issuecomment-1194651960 with some new insights

**Review by Lucxjo — APPROVED** [review]
All looks good! I agree with everything that davidzwa has said in previous comments.

Sorry it took a couple of days to get around to submitting this review. I've been a bit out of things recently

**Pull request merged**